go/ql/lib/semmle/go/Concepts.qll

```diff
@@ -355,8 +355,18 @@ module RegexpReplaceFunction {
  * extend `LoggerCall::Range` instead.
  */
 class LoggerCall extends DataFlow::Node instanceof LoggerCall::Range {
-  /** Gets a node that is a part of the logged message. */
-  DataFlow::Node getAMessageComponent() { result = super.getAMessageComponent() }
+  /**
+   * Gets a node whose value is a part of the logged message. Note that
+   * components corresponding to the format specifier "%T" are excluded as
+   * their type is logged rather than their value.
+   */
+  DataFlow::Node getAMessageComponent() {
```

Comment:
Suggesting keeping `getAMessageComponent` with the old semantics (i.e., get any formatted input, regardless of how it is formatted) and adding a new one, perhaps `getAValueFormattedMessageComponent` that excludes things formatted in such a way that disregards their value.